### DISCUSSION TITLE
Disable automatic hover?

### DISCUSSION THREAD

**Discussion by rgrinberg** [joaotavora/eglot#666] [Q&A] [answered]
Is it possible to disable automatic hover requests? I prefer to trigger hover manually as I find the echo area constantly flashing to be distracting.

**Comment by joaotavora** [comment, marked as answer]
Yes.  Have a look at `eglot-ignored-server-capabilities`.
```lisp
(defcustom eglot-ignored-server-capabilites (list)
  "LSP server capabilities that Eglot could use, but won't.
You could add, for instance, the symbol
`:documentHighlightProvider' to prevent automatic highlighting
under cursor."
  :type '(set
          :tag "Tick the ones you're not interested in"
          (const :tag "Documentation on hover" :hoverProvider)
          (const :tag "Code completion" :completionProvider)
          (const :tag "Function signature help" :signatureHelpProvider)
          (const :tag "Go to definition" :definitionProvider)
          (const :tag "Go to type definition" :typeDefinitionProvider)
          (const :tag "Go to implementation" :implementationProvider)
          (const :tag "Go to declaration" :implementationProvider)
          (const :tag "Find references" :referencesProvider)
          (const :tag "Highlight symbols automatically" :documentHighlightProvider)
          (const :tag "List symbols in buffer" :documentSymbolProvider)
          (const :tag "List symbols in workspace" :workspaceSymbolProvider)
          (const :tag "Execute code actions" :codeActionProvider)
          (const :tag "Code lens" :codeLensProvider)
          (const :tag "Format buffer" :documentFormattingProvider)
          (const :tag "Format portion of buffer" :documentRangeFormattingProvider)
          (const :tag "On-type formatting" :documentOnTypeFormattingProvider)
          (const :tag "Rename symbol" :renameProvider)
          (const :tag "Highlight links in document" :documentLinkProvider)
          (const :tag "Decorate color references" :colorProvider)
          (const :tag "Fold regions of buffer" :foldingRangeProvider)
          (const :tag "Execute custom commands" :executeCommandProvider)))
```

**Reply by rgrinberg** [reply]
Thanks for the pointer, but unless I'm misunderstanding this is not quite what I'm looking for. If I understood it correctly, your suggestion amounts to completely disabling hover functionality. I still want to use `textDocument/hover`, it's just that I want complete control over when this request is triggered. In particular, I want to:

* Disable eldoc completely
* Have an interactive function to trigger the hover where my cursor is.

**Reply by joaotavora** [reply]
To disable `eldoc` completely, just type `M-x eldoc-mode`  You'll lose some automatic diagnostics-showing functionality, but not all (the Flymake mode-line indicator still works, just the echo area stays put).

Now there is no interactive function to trigger just the hover where your cursor is.  But you _can_ call `eldoc` interactively, with `M-x eldoc`.  That will do what `eldoc-mode` normally does, but manually instead of automatically.  

Notice that `eldoc` doesn't only trigger hover.  It triggers all documentation-producing backends that are defined for Eglot (which are diagnostics, hover and signature info). 

It also pops a new window by default, but you can tweak `eldoc-display-functions` to control that.

I'm reluctant to add more functionality than this, to be honest.  You should have more or less what you're asking for with a couple of Elisp lines in your config.

**Reply by joaotavora** [reply]
@rgrinberg is this quesiton answered for you?  If so, please check it green.

**Comment by gety9** [comment]
@joaotavora 

If I add `:documentHighlightProvider to eglot-ignored-server-capabilites to disable auto highlighting, could i trigger highlight manually? Like eglot-highlight-at-point or smng like that? (Did M-x "highlight" but don't see any commands with eglot in auto suggestions)

**Reply by joaotavora** [reply]
> could i trigger highlight manually?

No.  Interesting feature though.  You may suggest it at bug-gnu-emacs@gnu.org

**Reply by gety9** [reply]
@joaotavora will do, thx
